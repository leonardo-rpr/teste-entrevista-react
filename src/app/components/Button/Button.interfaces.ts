export interface ButtonInterface {
  onClick: (event: any) => void;
}

export interface ButtonStylesInterface {
  onClick: (event: any) => void;
}
