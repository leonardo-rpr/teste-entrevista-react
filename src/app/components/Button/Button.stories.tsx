import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import Button from './Button';

storiesOf('Components', module)
  .addDecorator(withKnobs)
  .add('Button', () => (
    <div style={{ padding: '20px' }}>
      <Button loading={false} onClick={() => {}}>
        {text('Text', 'login')}
      </Button>
    </div>
  ));
