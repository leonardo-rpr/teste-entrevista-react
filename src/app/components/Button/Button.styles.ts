import styled from 'styled-components';
import { ButtonStylesInterface } from './Button.interfaces';

export const StyledButton = styled.button<ButtonStylesInterface>`
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: 'Montserrat', sans-serif;
  height: 48px;
  padding: 5px 20px;
  border-radius: 4px;
  font-size: 16px;
  font-weight: bold;
  text-transform: uppercase;
  border: none;
  outline: none;
  cursor: pointer;
  width: 100%;
  color: #ffffff;
  background: linear-gradient(to right, #f2295b 0%, #5a2b78 100%);
`;
