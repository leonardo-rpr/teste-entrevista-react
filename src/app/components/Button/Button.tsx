import React from 'react';
import { StyledButton } from './Button.styles';

import { ButtonInterface } from './Button.interfaces';

const Button: React.FC<ButtonInterface> = ({ children, onClick }) => (
  <StyledButton onClick={onClick}>{children}</StyledButton>
);

export default Button;
