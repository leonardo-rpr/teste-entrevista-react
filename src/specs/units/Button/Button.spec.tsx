import React from 'react';
import { render } from '@testing-library/react';
import Button from '../../../app/components/Button';
import 'jest-styled-components';

describe('Button', () => {
  it('Should render correctly', () => {
    const { container } = render(<Button onClick={jest.fn()}>Login</Button>);
    expect(container).toMatchSnapshot();
  });
});
