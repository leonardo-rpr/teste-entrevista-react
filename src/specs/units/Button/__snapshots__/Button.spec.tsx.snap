// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: 'Montserrat',sans-serif;
  height: 48px;
  padding: 5px 20px;
  border-radius: 4px;
  font-size: 16px;
  font-weight: bold;
  text-transform: uppercase;
  border: none;
  outline: none;
  cursor: pointer;
  width: 100%;
  color: #ffffff;
  background: linear-gradient(to right,#f2295b 0%,#5a2b78 100%);
}

<div>
  <button
    class="c0"
  >
    Login
  </button>
</div>
`;
